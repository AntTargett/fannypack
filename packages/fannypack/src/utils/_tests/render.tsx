import * as React from 'react';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (Component: any, { theme = {} } = {}) => render(<Provider theme={theme}>{Component}</Provider>);
