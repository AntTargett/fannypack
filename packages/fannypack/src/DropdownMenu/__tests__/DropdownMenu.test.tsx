import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { DropdownMenu } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test" visible>
        {menu => (
          <DropdownMenu {...menu} aria-label="Actions">
            <DropdownMenu.Item iconBefore="solid-pen" {...menu}>
              Edit
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-share" {...menu}>
              Share
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-file-signature" {...menu}>
              Rename
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-trash-alt" color="danger" {...menu}>
              Delete
            </DropdownMenu.Item>
          </DropdownMenu>
        )}
      </DropdownMenu.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for an invisible dropdown menu', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => (
          <DropdownMenu {...menu} aria-label="Actions">
            <DropdownMenu.Item iconBefore="solid-pen" {...menu}>
              Edit
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-share" {...menu}>
              Share
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-file-signature" {...menu}>
              Rename
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-trash-alt" color="danger" {...menu}>
              Delete
            </DropdownMenu.Item>
          </DropdownMenu>
        )}
      </DropdownMenu.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly for CSS props', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => (
          <DropdownMenu {...menu} aria-label="Actions" backgroundColor="red">
            <DropdownMenu.Item iconBefore="solid-pen" {...menu}>
              Edit
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-share" {...menu}>
              Share
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-file-signature" {...menu}>
              Rename
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-trash-alt" color="danger" {...menu}>
              Delete
            </DropdownMenu.Item>
          </DropdownMenu>
        )}
      </DropdownMenu.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('DropdownMenu.css.root should render correctly', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => (
          <DropdownMenu
            {...menu}
            overrides={{ DropdownMenu: { css: { root: { backgroundColor: 'red' } } } }}
            aria-label="Actions"
          >
            <DropdownMenu.Item iconBefore="solid-pen" {...menu}>
              Edit
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-share" {...menu}>
              Share
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-file-signature" {...menu}>
              Rename
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-trash-alt" color="danger" {...menu}>
              Delete
            </DropdownMenu.Item>
          </DropdownMenu>
        )}
      </DropdownMenu.State>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('DropdownMenu.css.root should render correctly', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => (
          <DropdownMenu {...menu} aria-label="Actions" backgroundColor="red">
            <DropdownMenu.Item iconBefore="solid-pen" {...menu}>
              Edit
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-share" {...menu}>
              Share
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-file-signature" {...menu}>
              Rename
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-trash-alt" color="danger" {...menu}>
              Delete
            </DropdownMenu.Item>
          </DropdownMenu>
        )}
      </DropdownMenu.State>,
      {
        // @ts-ignore
        theme: { DropdownMenu: { css: { root: { backgroundColor: 'red' } } } }
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(
      <DropdownMenu.State baseId="test">
        {menu => (
          <DropdownMenu {...menu} aria-label="Actions" backgroundColor="red">
            <DropdownMenu.Item iconBefore="solid-pen" {...menu}>
              Edit
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-share" {...menu}>
              Share
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-file-signature" {...menu}>
              Rename
            </DropdownMenu.Item>
            <DropdownMenu.Item iconBefore="solid-trash-alt" color="danger" {...menu}>
              Delete
            </DropdownMenu.Item>
          </DropdownMenu>
        )}
      </DropdownMenu.State>,
      {
        // @ts-ignore
        theme: { DropdownMenu: { defaultProps: { className: 'test', color: 'primary' } } }
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
