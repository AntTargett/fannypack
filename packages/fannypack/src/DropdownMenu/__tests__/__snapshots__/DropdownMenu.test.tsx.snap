// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-28 {
  padding: 0.5rem 0;
  min-width: 200px;
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02) !important;
}

.emotion-29 {
  background-color: white;
  border-radius: 4px;
  max-width: 350px;
  z-index: 19900410;
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02);
}

.emotion-29:focus {
  outline: none;
  box-shadow: #9b90f6 0px 0px 0px 3px;
}

.emotion-4 {
  background-color: unset;
  cursor: pointer;
  display: block;
  font-weight: 600;
  padding: 0.375rem 1rem;
  text-align: left;
  width: 100%;
  -webkit-transition: background-color 0.1s;
  transition: background-color 0.1s;
}

a.emotion-4 {
  color: unset;
  fill: unset;
  -webkit-text-decoration: unset;
  text-decoration: unset;
}

a.emotion-4:hover {
  color: unset;
  fill: unset;
}

.emotion-4[disabled] {
  cursor: not-allowed;
  opacity: 0.5;
}

.emotion-4:not(:disabled):focus {
  outline: unset;
  background-color: #f7f7f7;
}

.emotion-4:not(:disabled):hover {
  background-color: #f7f7f7;
}

.emotion-4:not(:disabled):focus:active,
.emotion-4:not(:disabled):hover:active {
  background-color: #f2f2f2;
}

.emotion-4 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-3 {
  margin-right: 0.5rem;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

.emotion-27 {
  color: #bb2a1a !important;
}

.emotion-31 {
  background-color: red !important;
  color: #3722ed !important;
}

.emotion-31 {
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02);
}

<div
  aria-label="Actions"
  aria-orientation="vertical"
  class="emotion-28 emotion-29 emotion-1 emotion-31 hidden test fp-Box fp-DropdownMenu"
  data-dialog="true"
  hidden=""
  id="test"
  role="menu"
  style="display: none;"
  tabindex="-1"
>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-1"
    role="menuitem"
    tabindex="0"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Edit
  </div>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-2"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Share
  </div>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-3"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Rename
  </div>
  <div
    class="emotion-4 emotion-1 emotion-27 fp-Box fp-DropdownMenuItem"
    id="test-4"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Delete
  </div>
</div>
`;

exports[`overrides DropdownMenu.css.root should render correctly 1`] = `
.emotion-31 {
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4 {
  background-color: unset;
  cursor: pointer;
  display: block;
  font-weight: 600;
  padding: 0.375rem 1rem;
  text-align: left;
  width: 100%;
  -webkit-transition: background-color 0.1s;
  transition: background-color 0.1s;
}

a.emotion-4 {
  color: unset;
  fill: unset;
  -webkit-text-decoration: unset;
  text-decoration: unset;
}

a.emotion-4:hover {
  color: unset;
  fill: unset;
}

.emotion-4[disabled] {
  cursor: not-allowed;
  opacity: 0.5;
}

.emotion-4:not(:disabled):focus {
  outline: unset;
  background-color: #f7f7f7;
}

.emotion-4:not(:disabled):hover {
  background-color: #f7f7f7;
}

.emotion-4:not(:disabled):focus:active,
.emotion-4:not(:disabled):hover:active {
  background-color: #f2f2f2;
}

.emotion-4 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-3 {
  margin-right: 0.5rem;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

.emotion-27 {
  color: #bb2a1a !important;
}

.emotion-28 {
  padding: 0.5rem 0;
  min-width: 200px;
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02) !important;
}

.emotion-28 {
  background-color: red;
}

.emotion-29 {
  background-color: white;
  border-radius: 4px;
  max-width: 350px;
  z-index: 19900410;
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02);
}

.emotion-29:focus {
  outline: none;
  box-shadow: #9b90f6 0px 0px 0px 3px;
}

.emotion-29 {
  background-color: red;
}

<div
  aria-label="Actions"
  aria-orientation="vertical"
  class="emotion-28 emotion-29 emotion-1 emotion-31 hidden fp-Box fp-DropdownMenu"
  data-dialog="true"
  hidden=""
  id="test"
  role="menu"
  style="display: none;"
  tabindex="-1"
>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-1"
    role="menuitem"
    tabindex="0"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Edit
  </div>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-2"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Share
  </div>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-3"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Rename
  </div>
  <div
    class="emotion-4 emotion-1 emotion-27 fp-Box fp-DropdownMenuItem"
    id="test-4"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Delete
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-31 {
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-28 {
  padding: 0.5rem 0;
  min-width: 200px;
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02) !important;
}

.emotion-29 {
  background-color: white;
  border-radius: 4px;
  max-width: 350px;
  z-index: 19900410;
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02);
}

.emotion-29:focus {
  outline: none;
  box-shadow: #9b90f6 0px 0px 0px 3px;
}

.emotion-4 {
  background-color: unset;
  cursor: pointer;
  display: block;
  font-weight: 600;
  padding: 0.375rem 1rem;
  text-align: left;
  width: 100%;
  -webkit-transition: background-color 0.1s;
  transition: background-color 0.1s;
}

a.emotion-4 {
  color: unset;
  fill: unset;
  -webkit-text-decoration: unset;
  text-decoration: unset;
}

a.emotion-4:hover {
  color: unset;
  fill: unset;
}

.emotion-4[disabled] {
  cursor: not-allowed;
  opacity: 0.5;
}

.emotion-4:not(:disabled):focus {
  outline: unset;
  background-color: #f7f7f7;
}

.emotion-4:not(:disabled):hover {
  background-color: #f7f7f7;
}

.emotion-4:not(:disabled):focus:active,
.emotion-4:not(:disabled):hover:active {
  background-color: #f2f2f2;
}

.emotion-4 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-3 {
  margin-right: 0.5rem;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

.emotion-27 {
  color: #bb2a1a !important;
}

<div
  aria-label="Actions"
  aria-orientation="vertical"
  class="emotion-28 emotion-29 emotion-1 emotion-31 fp-Box fp-DropdownMenu"
  data-dialog="true"
  id="test"
  role="menu"
  tabindex="-1"
>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-1"
    role="menuitem"
    tabindex="0"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Edit
  </div>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-2"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Share
  </div>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-3"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Rename
  </div>
  <div
    class="emotion-4 emotion-1 emotion-27 fp-Box fp-DropdownMenuItem"
    id="test-4"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Delete
  </div>
</div>
`;

exports[`props should render correctly for CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-28 {
  padding: 0.5rem 0;
  min-width: 200px;
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02) !important;
}

.emotion-29 {
  background-color: white;
  border-radius: 4px;
  max-width: 350px;
  z-index: 19900410;
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02);
}

.emotion-29:focus {
  outline: none;
  box-shadow: #9b90f6 0px 0px 0px 3px;
}

.emotion-4 {
  background-color: unset;
  cursor: pointer;
  display: block;
  font-weight: 600;
  padding: 0.375rem 1rem;
  text-align: left;
  width: 100%;
  -webkit-transition: background-color 0.1s;
  transition: background-color 0.1s;
}

a.emotion-4 {
  color: unset;
  fill: unset;
  -webkit-text-decoration: unset;
  text-decoration: unset;
}

a.emotion-4:hover {
  color: unset;
  fill: unset;
}

.emotion-4[disabled] {
  cursor: not-allowed;
  opacity: 0.5;
}

.emotion-4:not(:disabled):focus {
  outline: unset;
  background-color: #f7f7f7;
}

.emotion-4:not(:disabled):hover {
  background-color: #f7f7f7;
}

.emotion-4:not(:disabled):focus:active,
.emotion-4:not(:disabled):hover:active {
  background-color: #f2f2f2;
}

.emotion-4 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-3 {
  margin-right: 0.5rem;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

.emotion-27 {
  color: #bb2a1a !important;
}

.emotion-31 {
  background-color: red !important;
}

.emotion-31 {
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02);
}

<div
  aria-label="Actions"
  aria-orientation="vertical"
  class="emotion-28 emotion-29 emotion-1 emotion-31 hidden fp-Box fp-DropdownMenu"
  data-dialog="true"
  hidden=""
  id="test"
  role="menu"
  style="display: none;"
  tabindex="-1"
>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-1"
    role="menuitem"
    tabindex="0"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Edit
  </div>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-2"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Share
  </div>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-3"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Rename
  </div>
  <div
    class="emotion-4 emotion-1 emotion-27 fp-Box fp-DropdownMenuItem"
    id="test-4"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Delete
  </div>
</div>
`;

exports[`props should render correctly for an invisible dropdown menu 1`] = `
.emotion-31 {
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-28 {
  padding: 0.5rem 0;
  min-width: 200px;
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02) !important;
}

.emotion-29 {
  background-color: white;
  border-radius: 4px;
  max-width: 350px;
  z-index: 19900410;
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02);
}

.emotion-29:focus {
  outline: none;
  box-shadow: #9b90f6 0px 0px 0px 3px;
}

.emotion-4 {
  background-color: unset;
  cursor: pointer;
  display: block;
  font-weight: 600;
  padding: 0.375rem 1rem;
  text-align: left;
  width: 100%;
  -webkit-transition: background-color 0.1s;
  transition: background-color 0.1s;
}

a.emotion-4 {
  color: unset;
  fill: unset;
  -webkit-text-decoration: unset;
  text-decoration: unset;
}

a.emotion-4:hover {
  color: unset;
  fill: unset;
}

.emotion-4[disabled] {
  cursor: not-allowed;
  opacity: 0.5;
}

.emotion-4:not(:disabled):focus {
  outline: unset;
  background-color: #f7f7f7;
}

.emotion-4:not(:disabled):hover {
  background-color: #f7f7f7;
}

.emotion-4:not(:disabled):focus:active,
.emotion-4:not(:disabled):hover:active {
  background-color: #f2f2f2;
}

.emotion-4 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-3 {
  margin-right: 0.5rem;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

.emotion-27 {
  color: #bb2a1a !important;
}

<div
  aria-label="Actions"
  aria-orientation="vertical"
  class="emotion-28 emotion-29 emotion-1 emotion-31 hidden fp-Box fp-DropdownMenu"
  data-dialog="true"
  hidden=""
  id="test"
  role="menu"
  style="display: none;"
  tabindex="-1"
>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-1"
    role="menuitem"
    tabindex="0"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Edit
  </div>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-2"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Share
  </div>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-3"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Rename
  </div>
  <div
    class="emotion-4 emotion-1 emotion-27 fp-Box fp-DropdownMenuItem"
    id="test-4"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Delete
  </div>
</div>
`;

exports[`theming DropdownMenu.css.root should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-4 {
  background-color: unset;
  cursor: pointer;
  display: block;
  font-weight: 600;
  padding: 0.375rem 1rem;
  text-align: left;
  width: 100%;
  -webkit-transition: background-color 0.1s;
  transition: background-color 0.1s;
}

a.emotion-4 {
  color: unset;
  fill: unset;
  -webkit-text-decoration: unset;
  text-decoration: unset;
}

a.emotion-4:hover {
  color: unset;
  fill: unset;
}

.emotion-4[disabled] {
  cursor: not-allowed;
  opacity: 0.5;
}

.emotion-4:not(:disabled):focus {
  outline: unset;
  background-color: #f7f7f7;
}

.emotion-4:not(:disabled):hover {
  background-color: #f7f7f7;
}

.emotion-4:not(:disabled):focus:active,
.emotion-4:not(:disabled):hover:active {
  background-color: #f2f2f2;
}

.emotion-4 .fp-Icon {
  vertical-align: -0.125em;
}

.emotion-3 {
  margin-right: 0.5rem;
}

.emotion-0 {
  fill: inherit;
  position: relative;
  height: 1em;
  width: 1em;
}

.emotion-27 {
  color: #bb2a1a !important;
}

.emotion-31 {
  background-color: red !important;
}

.emotion-31 {
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02);
}

.emotion-28 {
  padding: 0.5rem 0;
  min-width: 200px;
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02) !important;
}

.emotion-28 {
  background-color: red;
}

.emotion-29 {
  background-color: white;
  border-radius: 4px;
  max-width: 350px;
  z-index: 19900410;
  box-shadow: 0px 4px 8px 0px rgba(0,0,0,0.1),0px 0px 4px 2px rgba(0,0,0,0.02);
}

.emotion-29:focus {
  outline: none;
  box-shadow: #9b90f6 0px 0px 0px 3px;
}

.emotion-29 {
  background-color: red;
}

<div
  aria-label="Actions"
  aria-orientation="vertical"
  class="emotion-28 emotion-29 emotion-1 emotion-31 hidden fp-Box fp-DropdownMenu"
  data-dialog="true"
  hidden=""
  id="test"
  role="menu"
  style="display: none;"
  tabindex="-1"
>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-1"
    role="menuitem"
    tabindex="0"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Edit
  </div>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-2"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Share
  </div>
  <div
    class="emotion-4 emotion-1 emotion-2 fp-Box fp-DropdownMenuItem"
    id="test-3"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Rename
  </div>
  <div
    class="emotion-4 emotion-1 emotion-27 fp-Box fp-DropdownMenuItem"
    id="test-4"
    role="menuitem"
    tabindex="-1"
  >
    <svg
      class="emotion-0 emotion-1 emotion-2 emotion-3 fp-DropdownMenuItemIcon fp-Box fp-Icon"
      role="img"
      viewBox="0 0 0 0"
    />
    Delete
  </div>
</div>
`;
